snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -297,20 +297,28 @@ def _ensure_compatible_build_environment(self) -> None:
     def _install_file(self, *, path: str, content: str, permissions: str) -> None:
         basename = os.path.basename(path)
 
-        with tempfile.NamedTemporaryFile(suffix=basename) as temp_file:
+        # Push to a location that can be written to by all backends
+        # with unique files depending on path.
+        # The path should be a valid path for the target.
+        remote_file = "/var/tmp/{}".format(base64.b64encode(path.encode()).decode())
+
+        # Windows cannot open the same file twice, so write to a temporary file that
+        # would later be deleted manually.
+        with tempfile.NamedTemporaryFile(delete=False, suffix=basename) as temp_file:
```

Comment:
I assume the file needed to be closed before multipass could push?

Reply:
correct